src/Runner.Common/JobServer.cs

```diff
@@ -56,6 +68,50 @@ public async Task ConnectAsync(VssConnection jobConnection)
             _hasConnection = true;
         }
 
+        private async Task CheckNetworkEndpointsAsync()
+        {
+            try
+            {
+                Trace.Info("Requesting Actions Service health endpoint status");
+                using (var httpClientHandler = HostContext.CreateHttpClientHandler())
+                using (var actionsClient = new HttpClient(httpClientHandler))
+                {
```

Comment:
do we want to set useragent for this request as well?

Reply:
Setting the user agent isn't required for ADN (unlike for GitHub), but yeah it'll be helpful in Kusto.